app/assets/javascripts/controllers/security_group/security_group_form_controller.js

```diff
@@ -24,9 +24,17 @@ ManageIQ.angular.app.controller('securityGroupFormController', ['securityGroupFo
       vm.afterGet = true;
       vm.modelCopy = angular.copy( vm.securityGroupModel );
 
+      supportedProviderTypes = [
+        'ManageIQ::Providers::Amazon::NetworkManager',
+        'ManageIQ::Providers::Azure::NetworkManager',
+        'ManageIQ::Providers::Google::NetworkManager',
+        'ManageIQ::Providers::Nuage::NetworkManager',
+        'ManageIQ::Providers::Openstack::NetworkManager',
+      ];
```

Comment:
hard-coding it here seems like the wrong approach.  Is there any way to ask for this list from the backend?
/cc @agrare @Fryguy

Reply:
Unfortunately, I don't think it's possible to do API query based on our `supports?` mechanism. Or even ask the API whether or not that particular entity supports a feature.

We can filter the allowed network provider types on API request level with `filter[]=type=ManageIQ::Providers::Google::NetworkManager` (and same for the others), but we'd still need to have the provider types hard-coded in the UI.

Reply:
I was pretty sure we had supports over the API via an OPTIONS call... @abellotti ?

Reply:
OPTIONS /api/providers would be preferred, if we don't have this info, need some enhancement here https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/providers_controller.rb#L171

Reply:
As far as I know, the API doesn't support this in general...
But specifically for ExtManagementSystem...

```
app/models/ext_management_system.rb
255:  virtual_column :supports_block_storage,  :type => :boolean
256:  virtual_column :supports_volume_multiattachment, :type => :boolean
257:  virtual_column :supports_volume_resizing, :type => :boolean
258:  virtual_column :supports_cloud_object_store_container_create, :type => :boolean
259:  virtual_column :supports_cinder_volume_types, :type => :boolean
260:  virtual_column :supports_volume_availability_zones, :type => :boolean
```

And that API can do :).
So we'd need to add the relevant virtual column to the model.

Reply:
We seem to have both `add_security_group` and `create_security_group` in `SupportsFeatureMixin`, but no ExtManagementSystem descendants use either of those. (There's also `supports_security_groups?` in a collector class, but not ems.)

Reply:
I'm removing the `create_security_group` feature here https://github.com/ManageIQ/manageiq/pull/20480 The `add_security_group` feature is used to add a security group to an OpenStack instance (i.e. it's not used to identify the capability to create the actual group).

Reply:
It looks like create_security_group is implemented by `Nsxt::NetworkManager`, `Openstack::NetworkManager`, and `Redhat::NetworkManager` (aka ovirt+openstack networking)

:+1: on not hard coding this, the `SecurityGroup` provider subclass has `supports :create` e.g. https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/app/models/manageiq/providers/openstack/network_manager/security_group.rb#L4 but if you need to ask the ExtManagementSystem then we'd need `supports :create_security_group`

Reply:
@agrare Yup, I agree. https://github.com/ManageIQ/manageiq/pull/20482 is implementing a new virtual attribute which will tell us whether the security group creation is possible or not. Will adjust this PR accordingly.